packages/toolkit/src/query/core/buildThunks.ts

```diff
@@ -435,37 +448,56 @@ In the case of an unhandled error, no tags will be "provided" or "invalidated".`
       endpointName: EndpointName,
       arg: any,
       options: PrefetchOptions
-    ): ThunkAction<void, any, any, AnyAction> =>
-    (dispatch: ThunkDispatch<any, any, any>, getState: () => any) => {
+    ): ThunkAction<PrefetchActionCreatorResult, any, any, AnyAction> =>
+    (dispatch: ThunkDispatch<any, any, AnyAction>, getState: () => any) => {
       const force = hasTheForce(options) && options.force
       const maxAge = hasMaxAge(options) && options.ifOlderThan
 
-      const queryAction = (force: boolean = true) =>
-        (api.endpoints[endpointName] as ApiEndpointQuery<any, any>).initiate(
-          arg,
-          { forceRefetch: force }
+      const dispatchPrefetchRequest = (
+        forceRefetch: boolean = true
+      ): PrefetchActionCreatorResult => {
+        const initiateOutput = dispatch(
+          (api.endpoints[endpointName] as ApiEndpointQuery<any, any>).initiate(
+            arg,
+            { forceRefetch, prefetch: options || true }
+          )
         )
-      const latestStateValue = (
-        api.endpoints[endpointName] as ApiEndpointQuery<any, any>
-      ).select(arg)(getState())
+
+        return {
+          unwrap() {
+            return initiateOutput.unwrap().then(noop, noop)
+          },
+          abort: initiateOutput.abort,
```

Comment:
what about `unsubscribe`? Why not just `return initiateOutput` as is?

Reply:
initiateOutput returns QueryActionCreatorResult

```ts
type QueryActionCreatorResult<
  D extends QueryDefinition<any, any, any, any>
> = Promise<QueryResultSelectorResult<D>> & {
  arg: QueryArgFrom<D>
  requestId: string
  subscriptionOptions: SubscriptionOptions | undefined
  abort(): void
  unwrap(): Promise<ResultTypeFrom<D>>
  unsubscribe(): void
  refetch(): void
  updateSubscriptionOptions(options: SubscriptionOptions): void
  queryCacheKey: string
}
```

It returns methods that do not fit the prefetch model like `updateSubscriptionOptions` and can cause more harm than anything else if misused.
I just want to expose the informations strictly useful.

Moreover we can always expose later more properties of initiate output if necessary, speaking of which we should at least expose `arg` and `queryCacheKey`.